File: ruby/test/client_test.rb
require "test/unit/runner/junitxml"

require "test_service_client"

module TestService
  class ClientTests < Test::Unit::TestCase
    def test_echo_body
      request_body = Message.new(int_field: 123, string_field: "the string")
      client = EchoClient.new(URI(ENV["SERVICE_URL"]))
      response = client.echo_body(body: request_body)
      assert_true response.respond_to? :ok
      assert_true response.ok?
      assert_equal request_body, response.ok
    end

    def test_echo_query_parameters
      client = EchoClient.new(URI(ENV["SERVICE_URL"]))
      response = client.echo_query(int_query: 123, string_query: "the string")
      assert_true response.respond_to? :ok
      assert_true response.ok?
      assert_equal OpenStruct.new(:ok => Message.new(int_field: 123, string_field: "the string"), :ok? => true), response
    end

    def test_echo_header
      client = EchoClient.new(URI(ENV["SERVICE_URL"]))
      response = client.echo_header(int_header: 123, string_header: "the string")
      assert_true response.respond_to? :ok
      assert_true response.ok?
      assert_equal Message.new(int_field: 123, string_field: "the string"), response.ok
    end

    def test_echo_url_params
      client = EchoClient.new(URI(ENV["SERVICE_URL"]))
      response = client.echo_url_params(int_url: 123, string_url: "value")
      assert_true response.respond_to? :ok
      assert_true response.ok?
      assert_equal Message.new(int_field: 123, string_field: "value"), response.ok
    end
  end
end

module TestService::V2
  class ClientTests < Test::Unit::TestCase
    def test_echo_body
      request_body = Message.new(bool_field: true, string_field: "the string")
      client = EchoClient.new(URI(ENV["SERVICE_URL"]))
      response = client.echo_body(body: request_body)
      assert_true response.respond_to? :ok
      assert_true response.ok?
      assert_equal request_body, response.ok
    end
  end
end

File: ts-node-fetch-superstruct/__tests__/client.test.ts
import { test } from 'uvu';
import * as assert from 'uvu/assert';

import {Message, Choice} from '../test-service/models'
import {client as echoClient} from '../test-service/echo'
import {client as checkClient} from '../test-service/check'

const config = {baseURL: process.env.SERVICE_URL!}

test('echoBody', async function() {
  const client = echoClient(config)
  let body: Message = {int_field: 123, string_field: "the string"}
  let response = await client.echoBody({body})
  assert.equal(response, body, 'response matches request')
})


test('echoQuery', async function() {
  const client = echoClient(config)
  let expected: Message = {int_field: 123, string_field: "the string"}
  let response = await client.echoQuery({intQuery: 123, stringQuery: "the string"})
  assert.equal(response, expected, 'response matches expected')
})

test('echoHeader', async function() {
  const client = echoClient(config)
  let expected: Message = {int_field: 123, string_field: "the string"}
  let response = await client.echoHeader({intHeader: 123, stringHeader: "the string"})
  assert.equal(response, expected, 'response matches expected')
})

test('echoUrlParams', async function() {
  const client = echoClient(config)
  let expected: Message = {int_field: 123, string_field: "the string"}
  let response = await client.echoUrlParams({intUrl: 123, stringUrl: "the string"})
  assert.equal(response, expected, 'response matches expected')
})

test('checkQuery', async function() {
  let client = checkClient(config)
  let body: Message = {int_field: 123, string_field: "the string"}
    await client.checkQuery({
        pString: "the string",
        pStringArray: ["string 1", "string 2"],
        pDate: "2021-01-01",
        pDateArray: ["2021-01-02"],
        pDatetime: new Date("2021-01-02T23:54"),
        pInt: 123,
        pLong: 123,
        pDecimal: 123,
        pEnum: Choice.SECOND_CHOICE,
        pStringOpt: "some string",
        pStringDefaulted: "the string",
    })
})

test.run();